pre-commit/action.yaml

```diff
@@ -18,22 +18,14 @@ runs:
       with:
         go-version: ^1.16
 
-    - name: Install pre-commit
-      run: |
-        python -m pip install pre-commit
-      shell: bash
-
     - name: Run pre-commit
-      run: |
-        pre-commit run -a --config ${{ inputs.pre-commit-config }}
-      shell: bash
+      uses: pre-commit/action@v3.0.0
+      with:
+        extra_args: --all-files --config ${{ inputs.pre-commit-config }}
+        token: ${{ steps.generate-token.outputs.token }}
 
     - name: Push pre-commit fixes
       if: always() # Push changes even when pre-commit fails
-      run: |
-        if ! git diff --quiet; then
-          git add -u
-          git commit -m "pre-commit fixes"
-          git push
-        fi
-      shell: bash
+      uses: stefanzweifel/git-auto-commit-action@v4
+      with:
+        commit_message: "ci(pre-commit): autofix"
```

Comment:
Maybe using `style` is better?
https://github.com/commitizen/conventional-commit-types/blob/d1fb9cc953b8f1e33e2dd63babdf6ae8ab1c8711/index.json#L15-L18

Or using `pre-commit fixes` to be the same as `pre-commit.ci` is acceptable.

```suggestion
        commit_message: "style(pre-commit): autofix"
```

Reply:
Thanks! I also think that is better.
But I referred this commit message from `pre-commit.cii`.
![image](https://user-images.githubusercontent.com/11865769/204949517-60be42b2-5d14-47c5-be51-f2dff2b153a0.png)
https://github.com/autowarefoundation/autoware.universe/pull/2333/commits/6eee79d4358add752ada35316eb72b774fb24dab

Reply:
Ah, I see. It's from https://github.com/autowarefoundation/autoware.universe/blob/8bbc5052588c7909e4fa52b6fdbcd6678f1afc75/.pre-commit-config.yaml#L2.

Maybe I made a mistake. I'll check why I used this message.

Reply:
It was modified from the example in https://pre-commit.ci/.
![image](https://user-images.githubusercontent.com/31987104/204950022-5726f4ff-905d-4d62-85be-7083ca6fba34.png)

Let's change `.pre-commit-config.yaml` as well!

Reply:
I fixed :smile: 
https://github.com/autowarefoundation/autoware-github-actions/pull/198/commits/04b6773d2836bc192610a3ef919c8f5deb77527e

Reply:
Thanks. For other repositories, I'll send PRs.

Reply:
https://github.com/autowarefoundation/autoware/pull/3094